SwiftWisdom/Core/StandardLibrary/Numbers/Mathable.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  Mathable.swift
+//  SwiftWisdom
+//
+//  Created by Paul Rolfe on 2/12/16.
+//  Copyright © 2016 Intrepid. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+public protocol Mathable: Comparable {
+    func + (lhs: Self, rhs: Self) -> Self
+    func / (lhs: Self, rhs: Self) -> Self
+    func * (lhs: Self, rhs: Self) -> Self
+    func - (lhs: Self, rhs: Self) -> Self
+    init(_ int: IntMax)
+}
+
+extension CollectionType where Generator.Element: Mathable {
+    
+    public func ip_average() -> Generator.Element? {
+        return ip_mean()
+    }
+    
+    public func ip_mean() -> Generator.Element? {
+        let zero = Generator.Element(0.toIntMax())
+        let count = Generator.Element(self.count.toIntMax())
+        guard count > zero else { return nil }
+        return reduce(zero, combine: +) / count
+    }
+    
+    public func ip_median() -> Generator.Element? {
+        let zero = Generator.Element(0.toIntMax())
+        let count = Generator.Element(self.count.toIntMax())
+        guard count > zero else { return nil }
+        return sort(<).ip_middleValues?.ip_mean()
+    }
+    
+    public func ip_mode() -> [Generator.Element]? {
+        return ip_mostCommonElements()
```

Comment:
I think we can pull it out and just leave mode for now, but there needs to be a check for no repeated values.  I believe if no value is repeated or takes precedence, then there is no mode.  It might make sense to do this with a dictionary and constrain to hashable to make it easier.  We should only be using numbers anyway so hashable should be easy to come by